moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1390,27 +1391,6 @@ bool PlanningScene::processAttachedCollisionObjectMsg(const moveit_msgs::Attache
       object.object.operation == moveit_msgs::CollisionObject::APPEND)
   {
     // STEP 0: Check message validity
-    if (object.object.primitives.size() != object.object.primitive_poses.size())
```

Comment:
That was to incorporate [this comment](https://github.com/ros-planning/moveit_msgs/pull/69#discussion_r415994724) from @v4hn, to allow not setting those poses if they are not required. This was a bit hasty though - instead of deleting the check, this should fill up the `poses` vector with identity poses to avoid access errors.